### ISSUE TITLE
Style/TopLevelMethodDefinition misses methods declared after a class.

### ISSUE THREAD

**Issue by nertzy** [rubocop/rubocop#9819]
See my [gist](https://gist.github.com/nertzy/47d18940864c3348d60a6897661522fa) for all relevant files and output.

## Expected behavior

`Style/TopLevelMethodDefinition` should detect top-level methods regardless of where they appear in a file, including after a class definition.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem

Turn on the `Style/TopLevelMethodDefinition` cop:

```yaml
Style/TopLevelMethodDefinition:
  Enabled: true
```

Run rubocop against the following file:

```ruby
class One
  def two
    2
  end
end

def one
  1
end
```

The top-level method `one` will not get detected.

Note that the cop works if the method is the only contents of the file:

```ruby
def one
  1
end
```

## RuboCop version

```
❯ bundle exec rubocop -V
1.15.0 (using Parser 3.0.1.1, rubocop-ast 1.5.0, running on ruby 2.7.3 x86_64-darwin20)
```
